### ISSUE TITLE
Allow release_automation.sh to be restarted

### ISSUE THREAD

**Issue by guarani** [wordpress-mobile/release-toolkit-gutenberg-mobile#8]
If aborted, `./release_automation.sh` cannot be re-run safely since it assumes an initial state where none of the steps the scripts performs is already done.
In practice this means that if I abort the script, I have limited options to finish the release:
- Manually clean up any work the script has already done and restart the script from the beginning, or
- Manually perform by hand the remaining steps that were not yet performed by the script

Ideally the script would determine the current release state before executing (e.g. if the gutenberg and gutenberg-mobile steps are already done, resume from the step where WPiOS and WPAndroid PRs are created).

An open question is whether the GitHub API will have all the information the script needs to check the release's current state (e.g. what PRs are open, what branches are made, etc).

**Comment by fluiddot** [comment]
Unfortunately I experienced this in a recent release, in my case I decided to cleanup the work that the script executed.

I think it would be nice to introduce some improvements to make the script re-run safely, here I added some potential improvements:

## Potential improvements
1. Before asking for the new version number, ask if the release process was already run.
2. Before creating the release branches, check if they already exists and if so ask the user to skip the creation and committing version update.
3. Before committing updates to Gutenberg submodule, check if there's any change if not skip it.
4. After generating the bundle, check if there's any change if not skip it.
5. Before creating a PR, check if it already exists and if so ask the user to skip the creation.
6. In WPiOS and WPAndroid steps make something similar about checking the branches and PRs.

**Comment by dcalhoun** [comment]
I agree that adding logic to allow the script to restart wherever it left off is the most ideal outcome. 

One alternative could be adding a `reset` command to the script that merely deletes the relevant local and remote release branches. From my experience, this would at least allow the script to be rerun. I do not think it's the most ideal solution to the problem of script run failures, but, if it's quicker to build a `reset` command, it might be a worthwhile iteration of the toolkit.

**Comment by fluiddot** [comment]
>One alternative could be adding a reset command to the script that merely deletes the relevant local and remote release branches. From my experience, this would at least allow the script to be rerun. I do not think it's the most ideal solution to the problem of script run failures, but, if it's quicker to build a reset command, it might be a worthwhile iteration of the toolkit.

Yeah, I agree it would be quicker to build the reset command than adding conditions to each step of the script to allow it to be restarted. The only concern I have is about the risk of executing the command accidentally and deleting an ongoing release but we could explicitly warn and ask confirmation when it gets executed.
